[NEW] Redesign authentication views

philipbrito opened the pull request:
Closes https://github.com/RocketChat/Rocket.Chat.Android/pull/1504 
Closes #1454

divyanshub024 committed:
onboarding screen

divyanshub024 committed:
connect server screen

divyanshub024 committed:
Merge branch 'develop' into redesignOnboarding

philipbrito committed:
Merge branch 'develop' into redesignOnboarding

philipbrito committed:
Merge branch 'v3.0.0' into redesignOnboarding

divyanshub024 committed:
login screen

divyanshub024 committed:
Merge branch 'develop-2.x' into redesignOnboarding

# Conflicts:
#	gradle/wrapper/gradle-wrapper.properties

divyanshub024 committed:
new screens

divyanshub024 committed:
Merge remote-tracking branch 'origin/redesignOnboarding' into redesignOnboarding

# Conflicts:
#	build.gradle

divyanshub024 committed:
join community

divyanshub024 committed:
undo

divyanshub024 committed:
new screens

divyanshub024 committed:
onboarding transaltion

divyanshub024 committed:
Merge remote-tracking branch 'origin/develop-2.x' into develop-2.x

divyanshub024 committed:
Merge branch 'develop-2.x' into redesignOnboarding

# Conflicts:
#	app/src/main/java/chat/rocket/android/authentication/login/presentation/LoginPresenter.kt
#	app/src/main/java/chat/rocket/android/authentication/login/presentation/LoginView.kt
#	app/src/main/java/chat/rocket/android/authentication/login/ui/LoginFragment.kt
#	app/src/main/java/chat/rocket/android/authentication/server/ui/ServerFragment.kt
#	app/src/main/java/chat/rocket/android/authentication/ui/AuthenticationActivity.kt
#	app/src/main/res/layout/fragment_authentication_log_in.xml
#	gradle/wrapper/gradle-wrapper.properties

divyanshub024 committed:
minor changes

divyanshub024 committed:
onboarding translation

divyanshub024 committed:
login option string translation

divyanshub024 committed:
login string translation

divyanshub024 committed:
sign up string translation

divyanshub024 committed:
2FA string translation

divyanshub024 committed:
Merge branch 'develop-2.x' into redesignOnboarding

# Conflicts:
#	app/src/main/java/chat/rocket/android/authentication/login/ui/LoginFragment.kt
#	app/src/main/java/chat/rocket/android/authentication/presentation/AuthenticationNavigator.kt
#	app/src/main/java/chat/rocket/android/authentication/twofactor/ui/TwoFAFragment.kt
#	app/src/main/java/chat/rocket/android/authentication/ui/AuthenticationActivity.kt
#	app/src/main/java/chat/rocket/android/profile/ui/ProfileFragment.kt
#	app/src/main/res/layout/fragment_authentication_log_in.xml
#	app/src/main/res/layout/fragment_profile.xml
#	app/src/main/res/menu/navigation.xml
#	app/src/main/res/values-de/strings.xml
#	app/src/main/res/values-es/strings.xml
#	app/src/main/res/values-fr/strings.xml
#	app/src/main/res/values-hi-rIN/strings.xml
#	app/src/main/res/values-pt-rBR/strings.xml
#	app/src/main/res/values-ru/strings.xml
#	app/src/main/res/values/strings.xml
#	build.gradle

divyanshub024 committed:
Merge branch 'develop-2.x' into redesignOnboarding

# Conflicts:
#	gradle/wrapper/gradle-wrapper.properties

divyanshub024 committed:
resolve conflicts

divyanshub024 committed:
Merge branch 'v3.0.0' into redesignOnboarding

divyanshub024 committed:
update gradle

divyanshub024 committed:
Merge remote-tracking branch 'origin/redesignOnboarding' into redesignOnboarding

divyanshub024 committed:
Add hindi translation

divyanshub024 committed:
update gradle

divyanshub024 committed:
Show menu in the toolbar for "terms and service" and "privacy policy".

philipbrito committed:
Refactor on the OnBoardingFragment

Fix the rocket.chat icon being draw incorrect on on boarding screen.
Adds proper widget themes.
Correct status bar color is now displayed accordingly the fragment (as per our desing).
Removes unneeded code.

philipbrito committed:
Updates the sign in to your server view

philipbrito committed:
Merge branch 'v3.0.0' into redesignOnboarding

philipbrito committed:
Removes the alert dialog when changing from https to http protocol

philipbrito committed:
Updates the  ServerFragment

philipbrito committed:
Redesign authentication views.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/RocketChat/Rocket.Chat.Android?pullRequest=1700) <br/>All committers have signed the CLA.

philipbrito committed:
Fix login with deep link and activity lifecycle

philipbrito committed:
Removes unused resources.

philipbrito committed:
Revert "Removes unused resources."

This reverts commit 6dd1ec6e54f7be71903b8cb2ca142d2c30bbdf84.

philipbrito committed:
Removes unsed resources.

philipbrito committed:
Fix expand and collapse state for the accounts view.

philipbrito committed:
GSL is now invoked without needing to tap the key icon.

philipbrito committed:
Merge branch 'v3.0.0' into new/redesign-authentication-views

rafaelks reviewed (changes requested):
My review so far @filipedelimabrito:

1. WordPress icon is wrong.

![screenshot_20180927-175914](https://user-images.githubusercontent.com/551004/46174896-ac78cb00-c280-11e8-8ad8-8f28b855aa76.png)

2. Profile screen is empty.

![screenshot_20180927-180618](https://user-images.githubusercontent.com/551004/46174908-b4d10600-c280-11e8-86cc-0e7c78727eb9.png)

3. Can't scroll to see the connect button while typing.

![screenshot_20180927-175845](https://user-images.githubusercontent.com/551004/46174925-bb5f7d80-c280-11e8-90dc-2414bde1f4e9.png)

4. Google Smart Lock is super intrusive, doesn't allow me to hide it or go back.

![screenshot_20180927-175938](https://user-images.githubusercontent.com/551004/46174943-c31f2200-c280-11e8-90b8-c883a2f0cd8d.png)

philipbrito commented:
@rafaelks Please note that this is a **WIP** PR.
For your point number four: Are we back again with the discussion about GSL? Btw, note that you can tap "NONE OF ABOVE" button to dismiss the dialog.

Please, let's review the UI (not the code) after the removing of the WIP on the title of the PR.
Thanks

rafaelks commented:
@filipedelimabrito Please check our `android-design` channel in order to understand how GSL should work. Also, "NONE OF ABOVE BUTTON" doesn't work. I can record a video for you if you want.

philipbrito commented:
@rafaelks Note that  you **CAN NOT** change the look and feel of the GSL! You're probably seeing that view because you have more than one account saved with GSL.

This is what I get for your reference:

![gsl](https://user-images.githubusercontent.com/11776825/46205465-13d16200-c2f7-11e8-871a-759788d178ab.jpg)

Again, since it is a **WIP PR** let's discuss anything **about the UI** internally. 
Thanks.

philipbrito committed:
Check for server accounts information before showing the login options view.

This also will show the login with username/email + password with there is only this method enabled by the server.

philipbrito committed:
Add wordpress icon.

philipbrito committed:
Adds scrolling to the server view.

philipbrito committed:
Updates the width of the spinner.

To make it better on some densities.

philipbrito committed:
Updates views.

philipbrito committed:
Fix the register button state.

philipbrito committed:
Fix bug that wasn't showing the login options for the first time.

philipbrito committed:
Fix getState not encoding correctly.

philipbrito committed:
Merge branch 'v3.0.0' into new/redesign-authentication-views

philipbrito committed:
Fix profile view not showing edit texts.

philipbrito committed:
Disables the GSL feature for now.

 To be improved soon.

rafaelks committed:
Merge branch 'develop' into new/redesign-authentication-views

philipbrito committed:
Merge branch 'develop' into new/redesign-authentication-views

philipbrito committed:
Merge branch 'develop' into new/redesign-authentication-views

philipbrito committed:
Add missing strings and remove the unneeded ones.

leonardoaramaki commented on app/src/main/java/chat/rocket/android/authentication/loginoptions/presentation/LoginOptionsPresenter.kt:
```diff
@@ -0,0 +1,187 @@
+                    when (loginType) {
+                        TYPE_LOGIN_OAUTH -> client.loginWithOauth(credentialToken, credentialSecret)
+                        TYPE_LOGIN_CAS -> {
+                            delay(3, TimeUnit.SECONDS)
```
Why we need this delay?

philipbrito replied:
We need to add a delay to make the CAS authentication work properly. If we remove that we'll see a message like "Unable to authenticate". This delay is needed due to the CAS authentication processing.

leonardoaramaki commented on app/src/main/java/chat/rocket/android/authentication/loginoptions/presentation/LoginOptionsPresenter.kt:
```diff
@@ -0,0 +1,187 @@
+                            client.loginWithCas(credentialToken)
+                        }
+                        TYPE_LOGIN_SAML -> {
+                            delay(3, TimeUnit.SECONDS)
```
Also here?

philipbrito replied:
We need to add a delay to make the SAML authentication work properly. If we remove that we'll see a message like "Unable to authenticate". This delay is needed due to the CAS authentication processing.

leonardoaramaki commented on app/src/main/java/chat/rocket/android/authentication/registerusername/ui/RegisterUsernameFragment.kt:
```diff
@@ -6,25 +6,39 @@ import android.os.Bundle
+private const val BUNDLE_USER_ID = "user_id"
+private const val BUNDLE_AUTH_TOKEN = "auth_token"
+
+fun newInstance(userId: String, authToken: String): Fragment {
```
Wouldn't be better if we put this method declaration in a `companion object` inside `RegisterUsernameFragment`?

philipbrito replied:
I'm not sure. `Companion object` works fine too, but doing that we'll need to make the same for all the others fragments.

leonardoaramaki commented on app/src/main/java/chat/rocket/android/authentication/resetpassword/ui/ResetPasswordFragment.kt:
```diff
@@ -108,42 +116,35 @@ class ResetPasswordFragment : Fragment(), ResetPasswordView {
     private fun disableUserInput() {
-        button_reset_password.isEnabled = false
+        disableButtonConnect()
         text_email.isEnabled = true
```
Is this always `true`?

philipbrito replied:
👀 👁  Thanks for the catch!
👍

leonardoaramaki commented on app/src/main/java/chat/rocket/android/authentication/server/ui/ServerFragment.kt:
```diff
@@ -188,47 +216,49 @@ class ServerFragment : Fragment(), ServerView {
-            val serverUrl = url.toString().removePrefix("${url.scheme()}://")
-            text_server_url.textContent = serverUrl
+            protocol = "${url.scheme()}://"
+            text_server_url.textContent = url.toString().removePrefix("${url.scheme()}://")
```
I think you meant to put `text_server_url.textContent = url.toString().removePrefix(protocol)` :thinking:

philipbrito replied:
Good catch!!
👍

philipbrito committed:
text_email should be disabled.

philipbrito committed:
Updates ServerFragment.kt

philipbrito committed:
Updates the actionmode color.
